frontend/ui/uimanager.lua

```diff
@@ -507,7 +507,15 @@ function UIManager:setDirty(widget, refreshtype, refreshregion, refreshdither)
                 end
             end
         elseif not widget.invisible then
-            self._dirty[widget] = true
+            -- We only ever check the dirty flag on top-level widgets, so only set it there!
+            for i = 1, #self._window_stack do
+                if self._window_stack[i].widget == widget then
+                    self._dirty[widget] = true
+                end
+            end
+            if not self._dirty[widget] then
+                logger.dbg("Tried to set a dirty flag on a non-toplevel widget", widget and (widget.name or widget.id or tostring(widget)) or "nil")
```

Comment:
How this will do in that case?
- FM with mosaic cover menu, 9 books with covers not yet fetched
- quick Hold on the first one > Hold menu (Cut|Copy|Paste...)
- covers thumbnails (no now more top leve) get fetched and updated

Reply:
It works, everything is dithered when needed, and nothing complains :)